src/Vision/IdentityDocumentParser.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Vision;
+
+use App\Entity\Adherent;
+use Cocur\Slugify\Slugify;
+
+class IdentityDocumentParser
+{
+    private $slugify;
+
+    public function __construct(Slugify $slugify)
+    {
+        $this->slugify = $slugify;
+    }
+
+    public function match(Adherent $adherent, ImageAnnotations $imageAnnotations): bool
+    {
+        return $this->hasFirstName($imageAnnotations, $adherent->getFirstName())
+            && $this->hasLastName($imageAnnotations, $adherent->getLastName())
+            && $this->hasDateOfBirth($imageAnnotations, $adherent->getBirthdate())
+        ;
+    }
+
+    public function hasFirstName(ImageAnnotations $imageAnnotations, string $firstName): bool
+    {
+        if ($imageAnnotations->isFrenchNationalIdentityCard()) {
+            preg_match('/Prénom( )?\(s\)( )?:( )?(?<first_names>.+)\\n/', $imageAnnotations->getText(), $matches);
+
+            $firstNames = array_map(function (string $firstName) {
+                return $this->normalize(trim($firstName));
+            }, preg_split('/[\s,]+/', $matches['first_names'] ?? null));
+
+            return \in_array($this->normalize($firstName), $firstNames, true);
+        } elseif ($imageAnnotations->isFrenchPassport()) {
+            $payload = $this->normalize($imageAnnotations->getText());
+
+            return false !== strpos($payload, $this->normalize($firstName));
+        }
+
+        throw new \InvalidArgumentException(sprintf('Instance of "%s" is not a handled identity document type.', ImageAnnotations::class));
+    }
+
+    public function hasLastName(ImageAnnotations $imageAnnotations, string $lastName): bool
+    {
+        if ($imageAnnotations->isFrenchNationalIdentityCard()) {
+            preg_match('/Nom( )?:( )?(?<last_name>.+)\\n/', $this->text, $matches);
+
+            dump($this->normalize($lastName));
+            dump($matches['last_name'] ?? null);
+
+            return $this->normalize($matches['last_name'] ?? null) === $this->normalize($lastName);
+        } elseif ($imageAnnotations->isFrenchPassport()) {
+            $payload = $this->normalize($imageAnnotations->getText());
+
+            return false !== strpos($payload, $this->normalize($lastName));
+        }
+
+        throw new \InvalidArgumentException(sprintf('Instance of "%s" is not a handled identity document type.', ImageAnnotations::class));
+    }
+
+    public function hasDateOfBirth(ImageAnnotations $imageAnnotations, \DateTimeInterface $dateOfBirth): bool
+    {
+        if ($imageAnnotations->isFrenchNationalIdentityCard()) {
+            preg_match('/(?<birth_date>[\d]{2}[\. ][\d]{2}[\. ][\d]{4})/', $this->text, $matches);
```

Comment:
You can remove brakcets when there is only one allowed value
```suggestion
            preg_match('/(?<birth_date>\d{2}[\. ]\d{2}[\. ]\d{4})/', $this->text, $matches);
```